protocols/api/src/test/java/org/apache/james/protocols/api/AbstractProtocolTransportTest.java

```diff
@@ -81,6 +84,11 @@ public boolean isStartTLSSupported() {
                 throw new UnsupportedOperationException();
             }
             
+            @Override
+            public Optional<SSLSession> getSSLSession() {
+                throw new UnsupportedOperationException();

```

Comment:
Why not return Optional.empty ?

Reply:
I am following the pattern here, since the method is not used in the tests. Throwing "unsupported" is different from returning Optional.empty() which would make it appear as a regular non-SSL transport.